Exclude the explicit assembly references from the nuspec when generating the snupkg. Explicit assembly references are not relevant from a snupkg perspective

nkolev92 opened the pull request:
## Bug

Fixes: https://github.com/NuGet/Home/issues/7638
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: Exclude the explicit assembly references from the nuspec when generating the snupkg. Explicit assembly references are not relevant from a snupkg perspective. 
Refer to https://github.com/NuGet/Home/issues/7638#issuecomment-498380498 for more details. 

## Testing/Validation

Tests Added: Yes  
Reason for not adding tests:  
Validation:  Manual, automation

nkolev92 committed:
Exclude the explicit assembly references from the nuspec when generating the snupkg. Explicit assembly references are not relevant from a snupkg perspective.

dtivel commented on test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:
```diff
@@ -483,6 +484,94 @@ public void PackCommand_SymbolPackageWithNuspecFile()
+                    {
+                        "contentFiles/any/any/image.jpg",
+                        "contentFiles/cs/net45/code.cs",
+                        Path.Combine("lib", "uap10.0", "packageA.dll").Replace('\\', '/'),
```
Why not simply:

    "lib/uap10.0/packageA.dll"

The same question applies [below](https://github.com/NuGet/NuGet.Client/pull/2869/files#diff-894bd5b54d5ce2ca8b552a16a53c7d62R569).

dtivel commented on test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:
```diff
@@ -483,6 +484,94 @@ public void PackCommand_SymbolPackageWithNuspecFile()
+                var package = new PackageArchiveReader(path);
+                var symbolPackage = new PackageArchiveReader(symbolPath);
+                var files = package.GetFiles()
+                    .Where(t => !t.StartsWith("[Content_Types]") && !t.StartsWith("_rels") && !t.StartsWith("package") && !t.EndsWith("nuspec"))
```
Create a well-named helper method and call it here and [here](https://github.com/NuGet/NuGet.Client/pull/2869/files#diff-894bd5b54d5ce2ca8b552a16a53c7d62R553).

nkolev92 replied:
Created an extension method and used it everywhere throughout the class.

dtivel commented on test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:
```diff
@@ -483,6 +484,94 @@ public void PackCommand_SymbolPackageWithNuspecFile()
+                    workingDirectory,
+                    "pack packageA.nuspec -symbols -SymbolPackageFormat snupkg",
+                    waitForExit: true);
+                Assert.True(0 == r.Item1, r.Item2 + " " + r.Item3);
```
This is more readable:

    Assert.True(r.Success, r.AllOutput);

dtivel commented on test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:
```diff
@@ -483,6 +484,94 @@ public void PackCommand_SymbolPackageWithNuspecFile()
+                Util.CreateFile(
+                    Path.Combine(workingDirectory, "contentFiles", "any", "any"),
+                    "image.jpg",
+                    "");
```
Why `""` here but `string.Empty` above?  The same question applies to line 513.  Perhaps explain the argument with a named parameter.

dtivel commented on test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:
```diff
@@ -483,6 +484,94 @@ public void PackCommand_SymbolPackageWithNuspecFile()
         }
 
+        [Fact]
+        public void PackCommand_SymbolPackageWithNuspecFileAndReference()
```
It would be good to express the expected outcome in the test name:

    <MethodUnderTest>_<Conditions>_<Expectation>

For example:

    PackCommand_WhenPackingSymbolsPackage_ExcludesAssemblyReferences

dtivel commented on test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:
```diff
@@ -483,6 +484,94 @@ public void PackCommand_SymbolPackageWithNuspecFile()
+                var path = Path.Combine(workingDirectory, "packageA.1.0.0.nupkg");
+                var symbolPath = Path.Combine(workingDirectory, "packageA.1.0.0.snupkg");
+                var package = new PackageArchiveReader(path);
+                var symbolPackage = new PackageArchiveReader(symbolPath);
```
Both `package` and `symbolPackage` are disposable.  Wrap with `using` statements.

nkolev92 commented:
@dtivel That test is merely a copy of another one with certain conditions changed. 
The pattern followed is exactly as the test above it. It just allowed me to iterate on this quicker. 
If/when the CI passes, I will go through those tests and refactor them.

zivkan commented on src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs:
```diff
@@ -905,6 +905,9 @@ private void BuildSymbolsPackage(string path)
                 symbolsBuilder.PackageTypes.Add(PackageType.SymbolsPackage);
+                symbolsBuilder.PackageAssemblyReferences.Clear();
+                // Remove the references when building the symbols package.
+                // They are not relevant for the symbols packages (snupkgs specifically).
```
comments are usually above the code they refer to. Any particular reason to break convention here?

nkolev92 replied:
I can move it above.

nkolev92 committed:
address feedback

nkolev92 commented:
@zivkan @dtivel 

Addressed all of your feedback. 
@dtivel I refactored the tests so that everything is properly disposed in all scenario in that test suite.

nkolev92 committed:
create an extesnion method

nkolev92 committed:
address more feedback
